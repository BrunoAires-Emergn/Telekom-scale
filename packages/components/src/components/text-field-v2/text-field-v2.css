/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

 scale-text-field-v2 {
  --transition: all var(--scl-motion-duration-fast)
    var(--scl-motion-easing-standard);
  --radius: var(--scl-radius-8);
  --border: var(--scl-spacing-1) solid var(--scl-color-text-standard);
  --border-error: var(--scl-spacing-2) solid var(--scl-color-background-error);
  --border-color-hover: var(--scl-color-primary-hover, #f90984);
  --border-color-focus: var(--scl-color-primary-hover, #f90984);
  --box-shadow-focus: 0 0 0 var(--scl-spacing-2) var(--scl-color-focus);
  --height: var(--scl-spacing-48);
  --height-small: var(--scl-spacing-40);
  --spacing-x: var(--scl-spacing-12);
  --color-disabled: var(--scl-color-background-disabled);
  --background-disabled: var(--scl-color-white);
  --border-color-readonly: var(--scl-color-grey-0);
  --background-readonly: var(--scl-color-grey-0);

  /*meta*/
  --font-weight-meta: var(--scl-font-weight-bold);
  --font-size-meta: var(--scl-font-size-12);
  --line-height-meta: var(--scl-font-line-height-133);
  --spacing-y-meta: var(--scl-spacing-4);
  --color-meta: var(--scl-color-text-standard);
  --color-meta-error: var(--scl-color-text-error);

  /*control*/
  --spacing-control: var(--scl-spacing-12) var(--spacing-x) 0
    calc(var(--spacing-x) - 1px);
  --transition-control: var(--transition);
  --font-size-control: var(--scl-font-size-16);

  /*counter*/
  --transition-counter: var(--transition);
  --font-size-counter: var(--font-size-meta);
  --line-height-counter: var(--line-height-meta);
  --color-counter-error: var(--color-meta-error);

  /*helper-text*/
  --transition-helper-text: var(--transition);
  --font-size-helper-text: var(--font-size-meta);
  --line-height-helper-text: var(--line-height-meta);
  --color-helper-text: var(--scl-color-blue-70);
  --color-helper-text-error: var(--color-meta-error);

  /*placeholder*/
  --transition-placeholder: var(--transition);
  --color-placeholder: var(--scl-color-grey-60);

  /*label*/
  --color-label: var(--scl-color-grey-60);
  --color-label-readonly: var(--scl-color-text-standard);
  --z-index-label: var(--scl-z-index-10);
  --transition-label: var(--transition);
  --font-size-label: var(--scl-font-size-16);
  --font-weight-label: var(--scl-font-weight-medium);
  --font-size-label-small: var(--scl-font-size-16);
  --font-weight-label-small: var(--scl-font-weight-medium);
  --font-size-label-focus: var(--scl-font-size-10);
  --font-weight-label-focus: var(--scl-font-weight-bold);
}

.text-field-v2 {
  position: relative;
}
.text-field-v2 .text-field-v2__helper-text,
.text-field-v2 .text-field-v2__counter {
  font-weight: var(--font-weight-meta);
}
.text-field-v2 .text-field-v2__control {
  width: 100%;
  height: var(--height);
  margin: 0;
  display: flex;
  outline: none;
  padding: var(--spacing-control);
  z-index: 1;
  box-sizing: border-box;
  transition: var(--transition-control);
  font-family: inherit;
  font-size: var(--font-size-control);
  border-radius: var(--radius);
  border: var(--border);
}

.text-field-v2.text-field-v2--has-prefix .text-field-v2__control {
  padding-left: calc(var(--spacing-x) + var(--prefix-length))
}

.text-field-v2.text-field-v2--has-suffix .text-field-v2__control {
  padding-right: calc(var(--spacing-x) - 1px + var(--suffix-length))
}


.text-field-v2 .text-field-v2__counter {
  display: flex;
  transition: var(--transition-counter);
  margin-left: auto;
  padding-right: var(--spacing-x);
  justify-content: flex-end;
  font-size: var(--font-size-counter);
  line-height: var(--line-height-counter);
  color: inherit;
}
.text-field-v2 .text-field-v2__helper-text {
  transition: var(--transition-helper-text);
  padding-left: var(--spacing-x);
  font-size: var(--font-size-helper-text);
  line-height: var(--line-height-helper-text);
  color: var(--color-helper-text);
}
.text-field-v2 .text-field-v2__meta {
  display: flex;
  justify-content: space-between;
  margin-top: var(--spacing-y-meta);
  color: var(--color-meta);
}

.text-field-v2:not(.text-field-v2--disabled):not(.text-field-v2--readonly)
  .text-field-v2__control:hover {
  border-color: var(--border-color-hover);
}

.text-field-v2:not(.text-field-v2--disabled):not(.text-field-v2--readonly)
  .text-field-v2__control:focus {
  border-color: var(--border-color-focus);
  box-shadow: var(--box-shadow-focus);
}

.text-field-v2:not(.text-field-v2--disabled) .text-field-v2__control:focus::placeholder {
  color: var(--color-placeholder);
}

.text-field-v2 .text-field-v2__control::placeholder,
.text-field-v2 ::placeholder {
  color: transparent;
  transition: var(--transition-placeholder);
}

.text-field-v2__label {
  top: 0;
  left: 0;
  color: var(--color-label);
  display: flex;
  z-index: var(--z-index-label);
  position: absolute;
  transition: var(--transition-label);
  pointer-events: none;
  font-size: var(--font-size-label);
  transform: translate(
    var(--spacing-x),
    calc((var(--height) - var(--font-size-label)) / 2)
  );
  font-weight: var(--font-weight-label);
}

.text-field-v2__prefix {
  top: var(--spacing-y-meta);
  left: 0;
  color: var(--color-label);
  display: flex;
  z-index: var(--z-index-label);
  position: absolute;
  transition: var(--transition-label);
  pointer-events: none;
  font-size: var(--font-size-label);
  transform: translate(
    var(--spacing-x),
    calc((var(--height) - var(--font-size-label)) / 2)
  );
  font-weight: var(--font-weight-label);
}

.text-field-v2__suffix {
  top: var(--spacing-y-meta);
  right: var(--scl-spacing-16);
  color: var(--color-label);
  display: flex;
  z-index: var(--z-index-label);
  position: absolute;
  transition: var(--transition-label);
  pointer-events: none;
  font-size: var(--font-size-label);
  transform: translate(
    var(--spacing-x),
    calc((var(--height) - var(--font-size-label)) / 2)
  );
  font-weight: var(--font-weight-label);
}

.text-field-v2--has-focus:not(.text-field-v2--readonly) .text-field-v2__label,
.animated .text-field-v2__label {
  line-height: var(--scl-font-variant-label-size);
  transform: translate(var(--spacing-x), var(--scl-spacing-8));
  font-size: var(--font-size-label-focus);
  font-weight: var(--font-weight-label-focus);
}
.text-field-v2--status-error .text-field-v2__control {
  border: var(--border-error);
}
.text-field-v2--status-error .text-field-v2__helper-text {
  color: var(--color-helper-text-error);
}
.text-field-v2--status-error .text-field-v2__counter {
  color: var(--color-counter-error);
}
.text-field-v2--size-small .text-field-v2__control {
  height: var(--height-small);
}
.text-field-v2--size-small .text-field-v2__label {
  font-size: var(--font-size-label-small);
  transform: translate(
    var(--spacing-x),
    calc((var(--height-small) - var(--font-size-label-small)) / 2)
  );
  font-weight: var(--font-weight-label-small);
}
.text-field-v2--size-small.text-field-v2--has-focus:not(.text-field-v2--readonly)
  .text-field-v2__label,
.text-field-v2--size-small.animated .text-field-v2__label,
.text-field-v2.text-field-v2--has-prefix .text-field-v2__label
{
  line-height: var(--scl-font-variant-label-size);
  transform: translate(var(--spacing-x), var(--scl-spacing-4));
  font-size: var(--font-size-label-focus);
}
.text-field-v2--transparent .text-field-v2__control {
  background-color: transparent;
}

.text-field-v2--readonly input,
.text-field-v2--readonly .text-field-v2__control {
  color: var(--color-label-readonly);
  border-color: var(--border-color-readonly);
  background: var(--background-readonly);
}

.text-field-v2--readonly .text-field-v2__control:focus {
  box-shadow: var(--box-shadow-focus);
}

.text-field-v2--disabled label,
.text-field-v2--disabled .text-field-v2__label,
.text-field-v2--disabled input,
.text-field-v2--disabled .text-field-v2__control,
.text-field-v2--disabled .text-field-v2__meta,
.text-field-v2--disabled .text-field-v2__counter,
.text-field-v2--disabled .text-field-v2__helper-text {
  cursor: not-allowed;
  border-color: var(--color-disabled);
  color: var(--color-disabled);
  background: var(--scl-color-white);
}
.text-field-v2--disabled.animated label.text-field-v2__label {
  color: var(--color-disabled);
}

/*Hide Arrows From Input Number*/

/* Chrome, Safari, Edge, Opera */
input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
}

/* Firefox */
input[type='number'] {
  -moz-appearance: textfield;
}
