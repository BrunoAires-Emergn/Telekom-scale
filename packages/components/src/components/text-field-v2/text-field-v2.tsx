/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

 import {
  Component,
  Prop,
  Element,
  Event,
  h,
  EventEmitter,
  Host,
  State,
} from '@stencil/core';
import classNames from 'classnames';
import { emitEvent } from '../../utils/utils';
import statusNote from '../../utils/status-note';

interface InputChangeEventDetail {
  value: string | number | boolean | undefined | null;
}

let i = 0;

@Component({
  tag: 'scale-text-field-v2',
  styleUrl: './text-field-v2.css',
  shadow: false,
})
export class TextFieldV2 {
  @Element() hostElement: HTMLElement;
  /** (optional) Input type */
  @Prop() type?:
    | 'email'
    | 'hidden'
    | 'number'
    | 'password'
    | 'tel'
    | 'text'
    | 'date'
    | 'url' = 'text';
  /** (optional) Input name */
  @Prop() name?: string = '';
  /** (optional) Input label */
  @Prop() label: string = '';
  /** (optional) Input size */
  @Prop() size?: string = '';
  /** (optional) Input helper text */
  @Prop() helperText?: string = '';
  /** @deprecated - invalid should replace status */
  @Prop() status?: string = '';
  /** (optional) Input status */
  @Prop() invalid?: boolean = false;
  /** (optional) Input max length */
  @Prop() maxLength?: number;
  /** (optional) Input min length */
  @Prop() minLength?: number;
  /** (optional) Input placeHolder */
  @Prop() placeholder?: string = '';
  /** (optional) Input disabled */
  @Prop() disabled?: boolean;
  /** (optional) Input readonly */
  @Prop() readonly?: boolean;
  /** (optional) Input required */
  @Prop() required?: boolean;
  /** (optional) Input counter */
  @Prop() counter?: boolean;
  /** (optional) Input value */
  @Prop({ mutable: true }) value?: string | number | null = '';
  /** (optional) Input checkbox id */
  @Prop() inputId?: string;
  /** (optional) input background transparent */
  @Prop() transparent?: boolean;
  /** (optional) the step attribute specifies the interval between legal numbers in an <input type="number"> element. */
  @Prop() step?: string = '1';
  /** (optional) input list */
  @Prop() list?: string;
  /** (optional) the input should automatically get focus when the page loads. */
  @Prop() inputAutofocus?: boolean;
  /** (optional) input prefix */
  @Prop() inputprefix?: string;
  /** (optional) input suffix */
  @Prop() inputsuffix?: string;

  /** (optional) Injected CSS styles */
  @Prop() styles?: string;

  /** Emitted when a keyboard input occurred. */
  @Event({ eventName: 'scale-input' }) scaleInput!: EventEmitter<KeyboardEvent>;
  /** @deprecated in v3 in favor of kebab-case event names */
  @Event({ eventName: 'scaleInput' })
  scaleInputLegacy!: EventEmitter<KeyboardEvent>;
  /** Emitted when the value has changed. */
  @Event({ eventName: 'scale-change' })
  scaleChange!: EventEmitter<InputChangeEventDetail>;
  /** @deprecated in v3 in favor of kebab-case event names */
  @Event({ eventName: 'scaleChange' })
  scaleChangeLegacy!: EventEmitter<InputChangeEventDetail>;
  /** Emitted when the input has focus. */
  @Event({ eventName: 'scale-focus' }) scaleFocus!: EventEmitter<void>;
  /** @deprecated in v3 in favor of kebab-case event names */
  @Event({ eventName: 'scaleFocus' }) scaleFocusLegacy!: EventEmitter<void>;
  /** Emitted when the input loses focus. */
  @Event({ eventName: 'scale-blur' }) scaleBlur!: EventEmitter<void>;
  /** @deprecated in v3 in favor of kebab-case event names */
  @Event({ eventName: 'scaleBlur' }) scaleBlurLegacy!: EventEmitter<void>;
  /** Emitted on keydown. */
  @Event({ eventName: 'scale-keydown' })
  scaleKeyDown!: EventEmitter<KeyboardEvent>;
  /** @deprecated in v3 in favor of kebab-case event names */
  @Event({ eventName: 'scaleKeydown' })
  scaleKeyDownLegacy!: EventEmitter<KeyboardEvent>;

  /** Whether the input element has focus */
  @State() hasFocus: boolean = false;

  componentWillLoad() {
    if (this.inputId == null) {
      this.inputId = 'input-text-field' + i++;
    }
  }

  componentDidRender() {
    if (this.status !== '') {
      statusNote({
        tag: 'deprecated',
        message:
          'Property "status" is deprecated. Please use the "invalid" property!',
        type: 'warn',
        source: this.hostElement,
      });
    }
    if (this.inputprefix) {
      const width = this.hostElement.querySelector('.text-field-v2__prefix').getBoundingClientRect().width
      this.hostElement.style.setProperty('--prefix-length',`${width}px`)
    }
    if (this.inputsuffix) {
      const width = this.hostElement.querySelector('.text-field-v2__suffix').getBoundingClientRect().width
      this.hostElement.style.setProperty('--suffix-length',`${width}px`)
    }    
  }

  // We're not watching `value` like we used to
  // because we get unwanted `scaleChange` events
  // because how we keep this.value up-to-date for type="select"
  // `this.value = selectedValue`

  emitChange() {
    emitEvent(this, 'scaleChange', {
      value: this.value == null ? this.value : this.value.toString(),
    });
  }

  handleInput = (event: Event) => {
    const target = event.target as HTMLInputElement | null;
    if (target) {
      this.value = target.value || '';
      this.emitChange();
    }
    emitEvent(this, 'scaleInput', event as KeyboardEvent);
  };

  handleChange = (event: Event) => {
    const target = event.target as HTMLInputElement | null;
    if (target) {
      this.value = target.value || '';
      this.emitChange();
    }
  };

  handleFocus = () => {
    emitEvent(this, 'scaleFocus');
    this.hasFocus = true;
  };

  handleBlur = () => {
    emitEvent(this, 'scaleBlur');
    this.hasFocus = false;
  };

  handleKeyDown = (event: KeyboardEvent) => {
    emitEvent(this, 'scaleKeyDown', event);
  };

  render() {
    const ariaInvalidAttr =
      this.status === 'error' || this.invalid ? { 'aria-invalid': true } : {};
    const helperTextId = `helper-message-${i}`;
    const ariaDescribedByAttr = { 'aria-describedBy': helperTextId };
    return (
      <Host>
        {this.styles && <style>{this.styles}</style>}
        <div class={this.getCssClassMap()}>
          {/* Accessibility: label should be always *before* the actual input */}
          <label class="text-field-v2__label" htmlFor={this.inputId}>
            {this.label}
          </label>
          {this.inputprefix !== '' ? <div class="text-field-v2__prefix"> {this.inputprefix} </div> : {}}          
          <input
            type={this.type}
            class="text-field-v2__control"
            value={this.value}
            {...(!!this.name ? { name: this.name } : {})}
            {...(!!this.inputAutofocus ? { autofocus: 'true' } : {})}
            required={this.required}
            minLength={this.minLength}
            maxLength={this.maxLength}
            id={this.inputId}
            list={this.list}
            onInput={this.handleInput}
            onChange={this.handleChange}
            onFocus={this.handleFocus}
            onBlur={this.handleBlur}
            onKeyDown={this.handleKeyDown}
            {...(!!this.placeholder ? { placeholder: this.placeholder } : {})}
            disabled={this.disabled}
            readonly={this.readonly}
            {...ariaInvalidAttr}
            {...(this.helperText ? ariaDescribedByAttr : {})}
            {...(this.type === 'number' ? { step: this.step } : {})}
          />

          {(!!this.helperText || !!this.counter) && (
            <div
              class="text-field-v2__meta"
              id={helperTextId}
              aria-live="polite"
              aria-relevant="additions removals"
            >
              {!!this.helperText && (
                <div class="text-field-v2__helper-text">{this.helperText}</div>
              )}
              {this.counter && (
                <div class="text-field-v2__counter">
                  {!!this.value ? String(this.value).length : 0} /{' '}
                  {this.maxLength}
                </div>
              )}
            </div>
          )}
          {this.inputsuffix && <div class="text-field-v2__suffix"> {this.inputsuffix} </div>}
        </div>
      </Host>
    );
  }

  getCssClassMap() {
    // input[type="date"] will print a placeholder in some browsers
    const animated =
      (this.value != null && this.value !== '') || this.type === 'date';

    return classNames(
      'text-field-v2',
      this.type && `text-field-v2--type-${this.type}`,
      this.hasFocus && 'text-field-v2--has-focus',
      this.disabled && `text-field-v2--disabled`,
      this.transparent && 'text-field-v2--transparent',
      this.status && `text-field-v2--status-${this.status}`,
      this.invalid && `text-field-v2--status-error`,
      this.size && `text-field-v2--size-${this.size}`,
      this.readonly && `text-field-v2--readonly`,
      this.inputprefix && `text-field-v2--has-prefix`,
      this.inputsuffix && `text-field-v2--has-suffix`,
      animated && 'animated'
    );
  }
}
