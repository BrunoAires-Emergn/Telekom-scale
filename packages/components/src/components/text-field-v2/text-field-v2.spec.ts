/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

 import { newSpecPage } from '@stencil/core/testing';
 import { TextFieldV2 } from './text-field-v2';
 
 describe('TextFieldV2', () => {
   let page;
   const TEST_VALUE = 'Test Value';
   beforeEach(async () => {
     page = await newSpecPage({
       components: [TextFieldV2],
       html: `<scale-text-field-v2></scale-text-field-v2>`,
     });
   });
 
   it('should match snapshot', async () => {
     expect(page.root).toMatchSnapshot();
   });
 
   it('should reflect attributes/props being set', async () => {
     const specPage = await newSpecPage({
       components: [TextFieldV2],
       html: `<scale-text-field-v2
                  type ="email"
                  name ="name"
                  label ="label"
                  max-length="2"
                  min-length="1"
                  size ="size"
                  invalid="true"
                  placeholder ="placeholder"
                  disabled ="true"
                  required ="true"
                  counter ="true" 
                  value ="value"
                  input-id ="input-text-field1"
                  transparent ="true"
                  styles ="styles"
                  list="datalist"
               ></scale-text-field-v2>`,
     });
 
     expect(specPage.rootInstance.type).toBe('email');
     expect(specPage.rootInstance.name).toBe('name');
     expect(specPage.rootInstance.label).toBe('label');
     expect(specPage.rootInstance.size).toBe('size');
     expect(specPage.rootInstance.maxLength).toBe(2);
     expect(specPage.rootInstance.minLength).toBe(1);
     expect(specPage.rootInstance.invalid).toBe(true);
     expect(specPage.rootInstance.placeholder).toBe('placeholder');
     expect(specPage.rootInstance.disabled).toBe(true);
     expect(specPage.rootInstance.required).toBe(true);
     expect(specPage.rootInstance.counter).toBe(true);
     expect(specPage.rootInstance.value).toBe('value');
     expect(specPage.rootInstance.transparent).toBe(true);
     expect(specPage.rootInstance.styles).toBe('styles');
     expect(specPage.rootInstance.inputId).toBe('input-text-field1');
     expect(specPage.rootInstance.hasFocus).toBe(false);
     expect(specPage.rootInstance.list).toBe('datalist');
   });
 
   it('should handle focus with right hasFocus state', async () => {
     const textField = new TextFieldV2();
     textField.hasFocus = false;
     textField.handleFocus();
     expect(textField.hasFocus).toBeTruthy();
   });
 
   it('should handle blur with right hasFocus state', async () => {
     const textField = new TextFieldV2();
     textField.hasFocus = true;
     textField.handleBlur();
     expect(textField.hasFocus).toBeFalsy();
   });
 
   it('should handle helperText prop', async () => {
     page.root.helperText = 'help';
     await page.waitForChanges();
     expect(page.rootInstance.helperText).toBe('help');
   });
 
   it('should handle maxLength prop', async () => {
     page.root.maxLength = 15;
     await page.waitForChanges();
     expect(page.rootInstance.maxLength).toBe(15);
   });
 
   it('should handle minLength prop', async () => {
     page.root.minLength = 5;
     await page.waitForChanges();
     expect(page.rootInstance.minLength).toBe(5);
   });
 
   it('should emit on input', async () => {
     const inputSpy = jest.fn();
     const inputSpyLegacy = jest.fn();
     page.doc.addEventListener('scale-input', inputSpy);
     page.doc.addEventListener('scaleInput', inputSpyLegacy);
     const inputField = page.doc.querySelector('input');
     inputField.value = TEST_VALUE;
     await inputField.dispatchEvent(new Event('input'));
     await page.waitForChanges();
     expect(inputSpy).toHaveBeenCalled();
     expect(inputSpyLegacy).toHaveBeenCalled();
   });
   it('should react on change event', async () => {
     // const changeSpy = jest.fn();
     // page.doc.addEventListener('scaleChange', changeSpy);
     const inputField = page.doc.querySelector('input');
     inputField.value = TEST_VALUE;
     await inputField.dispatchEvent(new Event('change'));
     await page.waitForChanges();
     expect(page.rootInstance.value).toBe('Test Value');
     inputField.value = '';
     await inputField.dispatchEvent(new Event('change'));
     await page.waitForChanges();
     expect(page.rootInstance.value).toBe('');
     // expect(changeSpy).toHaveBeenCalled();
   });
   it('should emit on change', async () => {
     const emitSpy = jest.fn();
     const emitSpyLegacy = jest.fn();
     page.doc.addEventListener('scale-change', emitSpy);
     page.doc.addEventListener('scaleChange', emitSpyLegacy);
     page.rootInstance.emitChange();
     await page.waitForChanges();
     expect(emitSpy).toHaveBeenCalled();
     expect(emitSpyLegacy).toHaveBeenCalled();
   });
   it('should emit on focus', async () => {
     const focusSpy = jest.fn();
     const focusSpyLegacy = jest.fn();
     page.doc.addEventListener('scale-focus', focusSpy);
     page.doc.addEventListener('scaleFocus', focusSpyLegacy);
     const inputField = page.doc.querySelector('input');
     inputField.value = TEST_VALUE;
     await inputField.dispatchEvent(new Event('focus'));
     await page.waitForChanges();
     expect(focusSpy).toHaveBeenCalled();
     expect(focusSpyLegacy).toHaveBeenCalled();
   });
   it('should emit on blur', async () => {
     const blurSpy = jest.fn();
     const blurSpyLegacy = jest.fn();
     page.doc.addEventListener('scale-blur', blurSpy);
     page.doc.addEventListener('scaleBlur', blurSpyLegacy);
     const inputField = page.doc.querySelector('input');
     inputField.value = TEST_VALUE;
     await inputField.dispatchEvent(new Event('blur'));
     await page.waitForChanges();
     expect(blurSpy).toHaveBeenCalled();
     expect(blurSpyLegacy).toHaveBeenCalled();
   });
   it('should emit on keydown', async () => {
     const keyDownSpy = jest.fn();
     const keyDownSpyLegacy = jest.fn();
     page.doc.addEventListener('scaleKeydown', keyDownSpyLegacy);
     page.doc.addEventListener('scale-keydown', keyDownSpy);
     const inputField = page.doc.querySelector('input');
     inputField.value = TEST_VALUE;
     await inputField.dispatchEvent(new Event('keydown'));
     await page.waitForChanges();
     expect(keyDownSpy).toHaveBeenCalled();
     expect(keyDownSpyLegacy).toHaveBeenCalled();
   });
 
   it('should handle css classes', () => {
     const element = new TextFieldV2();
 
     element.type = 'email';
     expect(element.getCssClassMap()).toContain('text-field-v2--type-email');
 
     element.hasFocus = true;
     expect(element.getCssClassMap()).toContain('text-field-v2--has-focus');
 
     element.disabled = true;
     expect(element.getCssClassMap()).toContain('text-field-v2--disabled');
 
     element.transparent = true;
     expect(element.getCssClassMap()).toContain('text-field-v2--transparent');
 
     element.invalid = true;
     expect(element.getCssClassMap()).toContain('text-field-v2--status-error');
 
     element.size = 'small';
     expect(element.getCssClassMap()).toContain('text-field-v2--size-small');
   });
 
   it('should handle hasFocus right', () => {
     const element = new TextFieldV2();
     element.handleFocus();
     expect(element.hasFocus).toBeTruthy();
     element.handleBlur();
     expect(element.hasFocus).toBeFalsy();
   });
   it('should render prefix and suffix', async () => {
    const specPage = await newSpecPage({
        components: [TextFieldV2],
        html: `<scale-text-field-v2
                   inputprefix="$"
                   inputsuffix="/100"
                   label="label"
                ></scale-text-field-v2>`,
    });
    const prefix = specPage.root.querySelector('.text-field-v2__prefix') as HTMLElement;
    const suffix = specPage.root.querySelector('.text-field-v2__suffix') as HTMLElement;
    expect(prefix.textContent.trim()).toBe('$')
    expect(suffix.textContent.trim()).toBe('/100')
  });
  it('should toggle password visibility', async () => {
    const specPage = await newSpecPage({
        components: [TextFieldV2],
        html: `<scale-text-field-v2
                   type ="password"
                   revealpassword=true
                   label ="label"
                ></scale-text-field-v2>`,
    });       
    expect(specPage.rootInstance.type).toBe('password');
    specPage.root.value = TEST_VALUE
    const revealButton = specPage.root.querySelector('.text-field-v2__reveal-password') as HTMLElement;
    revealButton.click();
    expect(specPage.rootInstance.type).toBe('text');
  });
  it('should reset the input value', async () => {
    const specPage = await newSpecPage({
        components: [TextFieldV2],
        html: `<scale-text-field-v2
                   type="text"
                   reset=true
                   label="label"
                ></scale-text-field-v2>`,
    });       
    specPage.root.value = TEST_VALUE
    const resetButton = specPage.root.querySelector('.text-field-v2__reset') as HTMLElement;
    resetButton.click();
    expect(specPage.root.value).toBe('');
  });  
 });
 