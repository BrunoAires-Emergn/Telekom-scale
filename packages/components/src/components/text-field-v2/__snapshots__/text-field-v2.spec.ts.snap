// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldV2 should match snapshot 1`] = `
<scale-text-field-v2>
  <div class="text-field-v2 text-field-v2--type-text">
    <scale-icon-undefined class="text-field-v2__prefix-icon"></scale-icon-undefined>
    <div class="text-field-v2__prefix"></div>
    <label class="text-field-v2__label" htmlfor="input-text-field-v20"></label>
    <input class="text-field-v2__control" id="input-text-field-v20" type="text" value="">
  </div>
</scale-text-field-v2>
`;
